File: jvm/leetcode_scala/src/main/scala/leetcode/editor/cn/[26]删除有序数组中的重复项.scala
package leetcode.editor.cn
object RemoveDuplicatesFromSortedArray {
//leetcode submit region begin(Prohibit modification and deletion)
  object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
      var slow = 0
      var fast = 0
      while (fast < nums.length) {
        if (nums(fast) != nums(slow)) {
          slow += 1
          nums(slow) = nums(fast)
        }
        fast += 1
      }
      slow + 1
    }
  }
//leetcode submit region end(Prohibit modification and deletion)

}

File: cpp/leetcodecn/[26]删除有序数组中的重复项.cpp
#include <vector>
#include "util.h"
#include <map>
#include <string>
#include <utility>
#include <memory>

using namespace std;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        if (nums.empty()) return 0;
        auto fast = nums.begin();
        auto slow = nums.begin();
        while (fast != nums.end()) {
            if (*fast != *slow) {
                slow++;
                *slow = *fast;
            }
            fast++;
        }

        slow++;
        return distance(nums.begin(), slow);
    }
};
//leetcode submit region end(Prohibit modification and deletion)


int main(int argv, char *argc[]) {
}